### ISSUE TITLE
How to run OSHi with IBM JVM?

### ISSUE THREAD

**Issue by zaisushe** [oshi/oshi#406]
When I run the following code, I will get a ClassCastException. I think it's caused the OSHi code is based on the Oracle JDK. I want to know is there a way to run OSHi based on IBM JDK?
Code:

 ```
 SystemInfo si = new SystemInfo();
	        HardwareAbstractionLayer hal = si.getHardware();
	        CentralProcessor  processor = hal.getProcessor();
	        processor.getSystemCpuLoad();
```

Exception:

`Exception in thread "main" java.lang.ClassCastException: com.ibm.lang.management.ExtendedOperatingSystem incompatible with com.sun.management.OperatingSystemMXBean`

**Comment by YoshiEnVerde** [comment]
I know I had that problem when checking OSHI on AIX...
I can't remember right now, but the fix was really simple.

I'll check my code at work and let you know if I can find it.
The problem isn't huge, just that the IBM JDL won't have the "com.sun" classes (as they are proprietary of Oracle)

**Comment by YoshiEnVerde** [comment]
Just to simplify my life, what's the OS you're running on? Could you post the full stacktrace of the error?

**Comment by zaisushe** [comment]
My application supposed to run on Windows, Mac and Linux. The  full stacktrace is:
```
Exception in thread "main" java.lang.ClassCastException: com.ibm.lang.management.ExtendedOperatingSystem incompatible with com.sun.management.OperatingSystemMXBean
	at oshi.hardware.common.AbstractCentralProcessor.initMXBean(AbstractCentralProcessor.java:123)
	at oshi.hardware.common.AbstractCentralProcessor.<init>(AbstractCentralProcessor.java:111)
	at oshi.hardware.platform.windows.WindowsCentralProcessor.<init>(WindowsCentralProcessor.java:72)
	at oshi.hardware.platform.windows.WindowsHardwareAbstractionLayer.getProcessor(WindowsHardwareAbstractionLayer.java:64)
	at com.spss.modeler.server.TestOSHi.main(TestOSHi.java:69)

```

As a workaround, I use following code to get the CPU usage for IBM JVM:
```
com.ibm.lang.management.OperatingSystemMXBean system =(com.ibm.lang.management.OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();		
		double rate=-1;		
		try {
			rate =system.getSystemCpuLoad(); //first call this method always return -1, and start the CPU monitor.
			Thread.sleep(500);//give some time for the monitor work.
			rate =system.getSystemCpuLoad();// get the CPU info for the last 500 mini seconds.
		} catch (InterruptedException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}	
```

**Comment by zaisushe** [comment]
I hope I could find a way to get the CPU usage from OSHi.

**Comment by YoshiEnVerde** [comment]
Ok, yes. The problem is a goof on our side... 
Line 123 of AbstractCentralProcessor does an explicit cast of the OperatingSystemMXBean object into the Sun specific one (which is not valid for the IBM JVM, that has its own implementation of that interface).

Even worse, I don't see any reason for the cast to be there...
@dbwiddis Do you know why that cast is there? Or can it be removed without problems?

**Comment by YoshiEnVerde** [comment]
Anyways, I'll test this and do a PR with a fix tomorrow

**Comment by dbwiddis** [comment]
@YoshiEnVerde that cast is inside a try / catch.  The intention is to test whether the Oracle MXBean is present and set variables appropriately to use its methods.  If it is not present, OSHI does backup calculations for things such as CPU load.

The correct fix is to add the `ClassCastException` to the catch block in addition to the existing `ClassNotFoundException`.

**Comment by dbwiddis** [comment]
Unrelated, if OSHI works on AIX, we should document that in the README, including any necessary steps. :)

**Comment by YoshiEnVerde** [comment]
@dbwiddis Good to know. Now that you pointed it out, I can see the flag being set based on the results of the check.
I'll add the missing exception to the catch block then

**Comment by dbwiddis** [comment]
Alternately you can try to detect the [IBM MXBean](https://www.ibm.com/support/knowledgecenter/en/SSYKE2_7.1.0/com.ibm.java.api.71.doc/com.ibm.lang.management/com/ibm/lang/management/OperatingSystemMXBean.html) and use it for AIX.  What would it take to make a full AIX port?

**Comment by YoshiEnVerde** [comment]
First of all: 
I've just made the PR for this fix: #408 

On the AIX issue:
> An AIX port would take **_a lot_** of effort.
> 
> At face value, AIX is a *nix clone/mod. However, the way it handles a lot of the hardware and OS info is very particular to the IBM proprietary machines it runs on.
> 
> That means we need to, pretty much, implement the port from scratch. It also means many of the things OSHI is currently reporting on are _just not there_ in AIX, or they require some heavy parsing and combining of command line call results.
> 
> I **will** get to implementing one, eventually, but right now it's just not feasible for me.

**Comment by YoshiEnVerde** [comment]
@zaisushe The fix has been merged. If you could test this, it should be working correctly now
